src/algorithms/toolbox.jl

```diff
@@ -240,14 +240,36 @@ end
 end
 
 """
-Adjoint of an MPS tensor, but permutes the physical spaces back into the codomain.
-Intuitively, this conjugates a tensor and then reinterprets its 'direction' as an MPS
-tensor.
+    edge_transfer_spectrum(top::Vector{E}, bot::Vector{E}; tol=Defaults.tol, num_vals=20,
+                           sector=one(sectortype(E))) where {E<:CTMRGEdgeTensor}
+
+Calculate the partial spectrum of the left edge transfer matrix corresponding to the given
+`top` vector of edges and a `bot` vector of edge. The `sector` keyword argument can be used
+to specify a non-trivial total charge for the transfer matrix eigenvectors. Specifically, an
+auxiliary space `ℂ[typeof(sector)](sector => 1)'` will be added to the domain of each
+eigenvector. The `tol` and `num_vals` keyword arguments are passed to `KrylovKit.eigolve`.
 """
-function _dag(A::MPSKit.GenericMPSTensor{S, N}) where {S, N}
-    return permute(A', ((1, (3:(N + 1))...), (2,)))
+function edge_transfer_spectrum(
+        top::Vector{E}, bot::Vector{E}; tol = MPSKit.Defaults.tol, num_vals = 20,
+        sector = one(sectortype(E))
+    ) where {E <: CTMRGEdgeTensor}
+    init = randn(
+        scalartype(E),
+        MPSKit._lastspace(first(bot))' ← ℂ[typeof(sector)](sector => 1)' ⊗ MPSKit._firstspace(first(top)),
+    )
+
+    transferspace = fuse(MPSKit._firstspace(first(top)) * MPSKit._lastspace(first(bot)))
+    num_vals = min(dim(transferspace, sector), num_vals) # we can ask at most this many values
+    eigenvals, eigenvecs, convhist = eigsolve(
+        flip(edge_transfermatrix(top, bot)), init, num_vals, :LM; tol = tol
+    )
+    convhist.converged < num_vals &&
+        @warn "correlation length failed to converge: normres = $(convhist.normres)"
+
+    return eigenvals
 end
```

Comment:
This is basically just a copy of `MPSKit.transfer_spectrum`, but using the dedicated edge transfer function instead. The main difference is that the bottom edge is a mandatory argument here (it simply can't be the same as the top since the physical spaces need to fit). There's quite a lot of duplication, but I don't know if it's worth it to really do anything about this.